packages/ui/src/multiplying-architecture/KaotoEditorChannelApi.ts

```diff
@@ -85,8 +85,9 @@ export interface KaotoEditorChannelApi extends KogitoEditorChannelApi {
 
   /**
    * Notifies the host application that a step was added.
+   * @param action The action performed on the step ('add', 'replace', 'remove').
    * @param stepType The type of the step that was added (e.g., "component", "processor", "entity").
    * @param stepName The name of the step that was added.
    */
-  onStepAdded(stepType: CatalogKind, stepName: string): Promise<void>;
+  onStepUpdated(action: StepUpdateAction, stepType: CatalogKind, stepName: string): Promise<void>;
 }
```

Comment:
@djelinek, this is how the new API would look like, please let me know what do you think.

Reply:
ah ok.. good to know